jgrapht-core/src/main/java/org/jgrapht/alg/intervalgraph/LexBreathFirstSearch.java

```diff
@@ -0,0 +1,217 @@
+package org.jgrapht.alg.intervalgraph;
+
+import org.jgrapht.*;
+
+import java.util.*;
+
+public class LexBreathFirstSearch<V, E>
+{
+    
+    /**
+     * Performs a lexicographical BFS starting at {@code startingVertex}.
+     *
+     * @param graph the graph we want to perform LBFS on
+     * @param startingVertex the starting vertex of the LBFS
+     * @return an array of vertices representing the order in which the vertices were found
+     */
+    public V[] lexBreathFirstSearch(Graph<V, E> graph, V startingVertex)
+    {
+        int n = graph.vertexSet().size();
+        ArrayList<V> ordering = new ArrayList<V>();
+        
+        String maxLabel = Integer.toString(n);
```

Comment:
Shouldnt this be updated somewhere in the loop further down? Have you tried the algorithm as is? Does it work?